bigtable-dataflow-import/src/main/java/com/google/cloud/bigtable/dataflowimport/DataCellPredicateFactory.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2015 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.bigtable.dataflowimport;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+import com.google.common.base.Function;
+import com.google.common.base.Predicate;
+
+import org.apache.hadoop.hbase.Cell;
+import org.apache.hadoop.hbase.CellComparator;
+import org.apache.hadoop.hbase.CellUtil;
+import org.apache.hadoop.hbase.KeyValue;
+
+/**
+ * Function that creates a {@link Predicate<Cell>}s from a Delete Marker {@code Cell}. The
+ * {@code Predicate} returns {@code true} for every cell that should be deleted because of the
+ * Delete Marker.
+ *
+ * <p>A predicate assumes that all cells to be filtered belong to the same row and column family
+ * as the seed Delete Marker, and does NOT validate this assumption on cells encountered.
```

Comment:
Yes. They are called from HBaseResultToMutationFn, where only one row is
handled at a time. That function separate cells by family before applying
the markers.
I should remove public as class qualifier.



On Tue, Feb 16, 2016 at 2:53 PM, Kevin Si <notifications@github.com> wrote:

> In
> bigtable-dataflow-import/src/main/java/com/google/cloud/bigtable/dataflowimport/DataCellPredicateFactory.java
> <https://github.com/GoogleCloudPlatform/cloud-bigtable-client/pull/667#discussion_r53068062>
> :
>
> > +
> > +import com.google.common.base.Function;
> > +import com.google.common.base.Predicate;
> > +
> > +import org.apache.hadoop.hbase.Cell;
> > +import org.apache.hadoop.hbase.CellComparator;
> > +import org.apache.hadoop.hbase.CellUtil;
> > +import org.apache.hadoop.hbase.KeyValue;
> > +
> > +/**
> > + * Function that creates a {@link Predicate<Cell>}s from a Delete Marker {@code Cell}. The
> > + * {@code Predicate} returns {@code true} for every cell that should be deleted because of the
> > + * Delete Marker.
> > + *
> > + * <p>A predicate assumes that all cells to be filtered belong to the same row and column family
> > + * as the seed Delete Marker, and does NOT validate this assumption on cells encountered.
>
> If the assumption is that row and column family should match, does it mean
> that some filtering must be done before the predicate can be used?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/cloud-bigtable-client/pull/667/files#r53068062>
> .
>